pupil_src/shared_modules/glfw.py

```diff
@@ -624,6 +628,21 @@ def glfwGetFramebufferSize(window):
     return width.value, height.value
 
 
+_Rectangle = collections.namedtuple("_Rectangle", ["x", "y", "width", "height"])
+_Margins = collections.namedtuple("_Margins", ["left", "top", "right", "bottom"])
```

Comment:
I feel like we could use `T.NamedTuple` subclasses here instead and also move `rectangle_intersection()` to `_Rectangle.intersection()`